driver-core/src/main/com/mongodb/internal/validator/ReplacingDocumentFieldNameValidator.java

```diff
@@ -34,13 +37,15 @@ public class ReplacingDocumentFieldNameValidator implements FieldNameValidator {
 
     @Override
     public boolean validate(final String fieldName) {
-        if (fieldName == null) {
```

Comment:
Removed this check since none of the other validators have it, and an assertion didn't seem worth it since the next line will throw anyway if fieldName is null.